src/components/target/TargetDetails.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+import { func, object, array, bool } from 'prop-types';
+import SmilesLogo from 'resources/icons/SmilesLogo';
+import TargetIcon from 'resources/icons/TargetIcon';
+import { deleteTarget, endSelectedTarget } from 'actions/targetActions';
+import MenuHeader from 'components/common/MenuHeader';
+import Modal from 'components/common/Modal';
+import Loading from 'components/common/Loading';
+
+class TargetDetails extends Component {
+  constructor() {
+    super();
+
+    this.confirmDelete = this.confirmDelete.bind(this);
+    this.closeDeleteModal = this.closeDeleteModal.bind(this);
+  }
+  state = { confirmDelete: false };
+
+  getTopicLabel(topicId) {
+    const { topics } = this.props;
+    let topic = null;
+
+    if (topics) {
+      topic = topics.find(topic => topic.topic.id == topicId);
+    }
+    return topic ? topic.topic.label : '';
+  }
+
+  getTopicIcon(topicId) {
```

Comment:
I saw that _getTopicIcon_ is already defined, maybe you could put it on a _utils_ class.